external-dns: allow custom sa (#1667)
giantswarm/docs: src/content/advanced/external-dns/gcp-custom-sa.md

---
linkTitle: Using a dedicated GCP service account
title: External DNS using a dedicated GCP service account

> description: This document describes how to configure External DNS to use a dedicated GCP Service Account.

Reviewer: ```suggestion
description: This document describes how to configure External DNS to use a dedicated GCP service account.
```

Should be same as in user questions, right?

weight: 25
menu:
  main:
    parent: advanced-external-dns
last_review_date: 2022-12-21
user_questions:
  - How can I override GCP service account used by External DNS?
owner:
  - https://github.com/orgs/giantswarm/teams/team-cabbage
---

## Overview

> By default, our External DNS App uses the `gcpProject` value to compose the right Service account annotation to correctly authenticate against GCP.

Reviewer: ```suggestion
By default, our External DNS App uses the `gcpProject` value to compose the right service account annotation to correctly authenticate against GCP.
```

I'd propose either `ServiceAccount` or just "service account".

Reviewer 2: In Google Cloud documentation, they refer to it as `service account`. Unless we use it differently, I'd vote we call it the same

Author: this refers to the Kubernetes service account, so I'd go for `ServiceAccount`

Author: ```suggestion
By default, our External DNS App uses the `gcpProject` value to compose the right annotation for the `ServiceAccount` to correctly authenticate against GCP.
```

## Configuration

> To override the default configuration, use the following values with the external-dns-app:

Reviewer: ```suggestion
To override the default configuration, use the following values with the `external-dns-app`:
```

... or just "External DNS App" to have it consistent with other mentions, your choice.
